lecture_3/03_project/calculator/internal/engine/handler/handler.go

```diff
@@ -0,0 +1,94 @@
+package handler
+
+import (
+	"context"
+	domainmodels "github.com/superbet-group/code-cadets-2021/lecture_3/03_project/calculator/internal/domain/models"
+	rabbitmqmodels "github.com/superbet-group/code-cadets-2021/lecture_3/03_project/calculator/internal/infrastructure/rabbitmq/models"
+	"log"
+)
+
+type Handler struct {
+	betRepository CalcBetRepository
+}
+
+func New(betRepository CalcBetRepository) *Handler {
+	return &Handler{
+		betRepository: betRepository,
+	}
+}
+
+func (h *Handler) HandleBets(ctx context.Context, bets <-chan rabbitmqmodels.Bet ) {
+	go func() {
+		for bet := range bets {
+			log.Println("Processing bet, betId:", bet.Id, " with selectionId:", bet.SelectionId)
+
+			domainBet := domainmodels.Bet{
+				Id:                   bet.Id,
+				SelectionId:          bet.SelectionId,
+				SelectionCoefficient: bet.SelectionCoefficient,
+				Payment:              bet.Payment,
+			}
+
+			err := h.betRepository.InsertCalcBet(ctx, domainBet)
+			if err != nil {
+				log.Println("Failed to insert bet, error: ", err)
+				continue
+			}
+			log.Println("Inserted bet, betId: ", domainBet.Id, ", selectionId: ", domainBet.SelectionId)
```

Comment:
What if bet already exists? You will log `uniqueViolationError` (I have never used `sqllite`, but it probably works like that).

In that case you should not log anything, or you could inform application developer through logs about that occasion. (With different log message, something user friendly).
Or maybe you could update existing bet in the DB with new coefficient.